File: example-databinding/src/main/java/com/xwray/groupie/example/databinding/ColumnGroup.java
package com.xwray.groupie.example.databinding;

import androidx.annotation.NonNull;

import com.xwray.groupie.Group;
import com.xwray.groupie.GroupDataObserver;
import com.xwray.groupie.Item;
import com.xwray.groupie.databinding.BindableItem;

import java.util.ArrayList;
import java.util.List;

/**
 * A simple, non-editable, non-nested group of Items which displays a list as vertical columns.
 */
public class ColumnGroup implements Group {

    private List<BindableItem<?>> items = new ArrayList<>();

    public ColumnGroup(List<? extends BindableItem<?>> items) {
        for (int i = 0; i < items.size(); i++) {
            // Rearrange items so that the adapter appears to arrange them in vertical columns
            int index;
            if (i % 2 == 0) {
                index = i / 2;
            } else {
                index = (i - 1) / 2 + (int) (items.size() / 2f);
                // If columns are uneven, we'll put an extra one at the end of the first column,
                // meaning the second column's indices will all be increased by 1
                if (items.size() % 2 == 1) index++;
            }
            BindableItem<?> trackItem = items.get(index);
            this.items.add(trackItem);
        }
    }

    @Override
    public void registerGroupDataObserver(@NonNull GroupDataObserver groupDataObserver) {
        // no real need to do anything here
    }

    @Override
    public void unregisterGroupDataObserver(@NonNull GroupDataObserver groupDataObserver) {
        // no real need to do anything here
    }

    @NonNull
    @Override
    public BindableItem<?> getItem(int position) {
        return items.get(position);
    }

    @Override
    public int getPosition(@NonNull Item item) {
        return items.indexOf(item);
    }

    @Override public int getItemCount() {
        return items.size();
    }
}

File: example-viewbinding/src/main/java/com/xwray/groupie/example/viewbinding/ColumnGroup.kt
package com.xwray.groupie.example.viewbinding

import com.xwray.groupie.Group
import com.xwray.groupie.GroupDataObserver
import com.xwray.groupie.Item
import com.xwray.groupie.viewbinding.BindableItem
import java.util.*

/**
 * A simple, non-editable, non-nested group of Items which displays a list as vertical columns.
 */
class ColumnGroup(items: List<BindableItem<*>>) : Group {

    private val items: ArrayList<BindableItem<*>> = ArrayList()

    init {
        for (i in items.indices) {
            // Rearrange items so that the adapter appears to arrange them in vertical columns
            val index = if (i % 2 == 0) {
                i / 2
            } else {
                // If columns are uneven, we'll put an extra one at the end of the first column,
                // meaning the second column's indices will all be increased by 1
                (i - 1) / 2 + (items.size / 2f).toInt() + if (items.size % 2 == 1) 1 else 0
            }
            val trackItem = items[index]
            this.items.add(trackItem)
        }
    }

    override fun registerGroupDataObserver(groupDataObserver: GroupDataObserver) {
        // no real need to do anything here
    }

    override fun unregisterGroupDataObserver(groupDataObserver: GroupDataObserver) {
        // no real need to do anything here
    }

    override fun getItem(position: Int): BindableItem<*> = items[position]

    override fun getPosition(item: Item<*>): Int = items.indexOf(item)

    override fun getItemCount(): Int = items.size
}
